### DISCUSSION TITLE
Formula 'chicken' (scheme) doesn't link properly on Apple M1

### DISCUSSION THREAD

**Discussion by corderguy** [Homebrew/discussions#3268] [Everyday usage] [answered]
Steps to reproduce: `brew install chicken && csi -ne '(import (chicken tcp))'`
Expected output: successful install followed by nothing (a successful import)
Actual output: successful install followed by: [this error](https://gist.github.com/jcguy/76302ef2ad3df9605ec85f6697a0eee2)

```shell
>> uname -a
Darwin My-MBP.lan 21.4.0 Darwin Kernel Version 21.4.0: Fri Mar 18 00:46:32 PDT 2022; root:xnu-8020.101.4~15/RELEASE_ARM64_T6000 arm64
```
[`brew config`](https://gist.github.com/jcguy/582149c1f61202a072a82040491f6e16)

```shell
>> brew doctor
Your system is ready to brew.
```

```shell
>> otool -L (which csi)
/opt/homebrew/bin/csi:
        /opt/homebrew/Cellar/chicken/5.3.0/lib/libchicken.dylib (compatibility version 1.0.0, current version 1.0.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.100.3)
```

```shell
>> ls -lh /opt/homebrew/Cellar/chicken/5.3.0/lib/libchicken.dylib
-r--r--r--  1 corder  admin   3.6M May 13 13:58 /opt/homebrew/Cellar/chicken/5.3.0/lib/libchicken.dylib
```

I am able to fix this problem by symlinking `/opt/homebrew/lib/libchicken.dylib` (which is itself a symlink to `/opt/homebrew/Cellar/chicken/5.3.0/lib/libchicken.dylib`) to `/usr/local/lib/libchicken.dylib`. I was also able to fix this problem by setting the following envvar:
```shell
>> DYLD_LIBRARY_PATH=/opt/homebrew/lib/ csi -ne '(import (chicken tcp))'
>>
```
However, I am not familiar enough with the Homebrew formula or the Chicken build system to figure out how to fix this in a better way. Has anyone else experienced this issue?

**Comment by SMillerDev** [comment]
You need to ensure `/opt/homebrew/lib` is in your include path. On Intel Homebrew installs in a path that's in the include path by default but on ARM that's up to the user.

**Reply by corderguy** [reply]
Is this documented anywhere? Would it be possible for Homebrew to set up the include path on install?

**Reply by SMillerDev** [reply]
https://developer.apple.com/library/archive/documentation/DeveloperTools/Conceptual/DynamicLibraries/100-Articles/DynamicLibraryUsageGuidelines.html documents how to use dynamic libraries. And I'm sure whatever tool you're using has documentation for it too.

Since there's no one way to set up libraries Homebrew can't do this for you.

**Comment by carlocab** [comment]
I've opened Homebrew/homebrew-core#101481 to fix this.

**Comment by carlocab** [comment, marked as answer]
Should be fixed now. Just do `brew update && brew upgrade chicken`.

**Reply by corderguy** [reply]
Thank you! I thought it might have to do with rpath, but wasn't able to figure it out myself.
